src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -511,23 +511,23 @@ private void writeRecord(SinkRecord record) throws IOException {
     long expectedOffset = offset + recordCounter;
     if (offset == -1) {
       offset = record.kafkaOffset();
-    } else if (record.kafkaOffset() != expectedOffset) {
+    } else if (record.kafkaOffset() < expectedOffset) {
```

Comment:
I think the logic doesn't quite work based on how `recordCounter` is updated though. You're right that on the first message this works fine. But let's say you start a new `TopicPartitionWriter` at offset 100, but actually messages 100-199 have been compacted. So you see offset 200. This check will work for that first message, but then the next iteration you'll have incremented `recordCounter` by 1 and `expectedOffset will be 101. While it will allow valid messages through, there's a gap of messages from 102-200 that wouldn't be properly rejected. So I think we'd need to get rid of recordCounter and just update a variable with the last seen offset to validate against.

re: the config setting, I think we will need it if we can't guarantee bad data won't be allowed through accidentally. If we can figure out a version that works and handles that bug, I'm all for doing it without the config setting.